src/libraries/System.Net.Security/tests/FunctionalTests/TestHelper.cs

```diff
@@ -47,5 +48,137 @@ internal static (VirtualNetworkStream ClientStream, VirtualNetworkStream ServerS
+                    data[0] = 0x30; //SEQUENCE
+                    data[1] = 22;
+                    data[2] = 0x80;
+                    data[3] = 20;
+                    Buffer.BlockCopy(sha1.ComputeHash(intermedCertWithKey.PublicKey.EncodedKeyValue.RawData), 0, data, 4, 20);
+                    serverReq.CertificateExtensions.Add(new X509Extension(new Oid("2.5.29.35"), data, false));
+                }
+
+                // 1.3.6.1.5.5.7.1.1
+                if (caUrl != null)
+                {
+                    var urlBytes = Encoding.ASCII.GetBytes(caUrl);
+
+                    byte[] data = new byte[urlBytes.Length + 16];
+                    data[0] = 0x30; //SEQUENCE
+                    data[1] = (byte)(urlBytes.Length + 14);
+                    data[2] = 0x30; //SEQUENCE;
+                    data[3] = (byte)(urlBytes.Length + 12);
+                    data[4] = 6; // OBJECT
+                    data[5] = 8; // LENGTH
+                    // OID
+                    data[6] = 0x2b;
+                    data[7] = 0x6;
+                    data[8] = 0x1;
+                    data[9] = 0x5;
+                    data[10] = 0x5;
+                    data[11] = 0x7;
+                    data[12] = 0x30; // SEQUENCE
+                    data[13] = 02;
+                    data[14] = 0x86;
+                    data[15] = (byte)(urlBytes.Length);
+                    data[16] = 0x74;
+                    data[17] = 0x74;
+                    data[18] = 0x70;
+                    Buffer.BlockCopy(urlBytes, 0, data, 16, urlBytes.Length);
+
+                    serverReq.CertificateExtensions.Add(new X509Extension(new Oid("1.3.6.1.5.5.7.1.1"), data, false));
+                }
+
+                serverReq.CertificateExtensions.Add(
+                    new X509KeyUsageExtension(
+                        X509KeyUsageFlags.DigitalSignature | X509KeyUsageFlags.KeyEncipherment | X509KeyUsageFlags.DataEncipherment,
+                        false));
+                serverReq.CertificateExtensions.Add(
+                    new X509EnhancedKeyUsageExtension(
+                        new OidCollection()
+                        {
+                            new Oid("1.3.6.1.5.5.7.3.1", null),
+                        },
+                        false));
+
+                SubjectAlternativeNameBuilder builder = new SubjectAlternativeNameBuilder();
+                builder.AddDnsName(name);
+                builder.AddIpAddress(IPAddress.Loopback);
+                builder.AddIpAddress(IPAddress.IPv6Loopback);
+                serverReq.CertificateExtensions.Add(builder.Build());
+
+                RandomNumberGenerator.Fill(serial);
+
+                X509Certificate2 serverCert = serverReq.Create(intermedCertWithKey, startTime, endTime, serial);
+                X509Certificate2 serverCertWithKey = serverCert.CopyWithPrivateKey(server);
+
+                chain.Add(intermedCertWithKey);
+                chain.Add(rootCertWithKey);
```

Comment:
To be more realistic, you should forget the CA private keys by the end of this method, since the TLS Server (or TLS Client) doesn't normally have access to those.